http/middleware.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Middleware handler based on {@linkcode Deno.ServeHandlerInfo} but an added
+ * `next()` function which calls the next middleware handler in the middleware
+ * chain.
+ */
+export type MiddlewareHandler = (
+  request: Request,
+  info: Deno.ServeHandlerInfo,
+  next: () => Response | Promise<Response>,
+) => Response | Promise<Response>;
+
+/**
+ * Creates a {@linkcode Deno.ServeHandler} from the given middleware chain,
+ * which can then be passed to {@linkcode Deno.serve}.
+ *
+ * @param middlewares Middleware chain
+ *
+ * @example
+ * ```ts
+ * import {
+ *   type MiddlewareHandler,
+ *   createHandler,
+ * } from "https://deno.land/std@$STD_VERSION/http/middleware.ts";
+ *
+ * const middleware1: MiddlewareHandler = async (_request, _info, next) => {
+ *   const start = performance.now();
+ *   const response = await next();
+ *   const duration = performance.now() - start;
+ *   response.headers.set("x-request-time", duration.toString());
+ *   return response;
+ * };
+ *
+ * const middleware2: MiddlewareHandler = (request, info) => {
+ *   return Response.json({ request, info });
+ * };
+ *
+ * const handler = createHandler([middleware1, middleware2])
+ * ```
+ */
+export function createHandler(
+  middlewares: MiddlewareHandler[],
+): Deno.ServeHandler {
+  return (request, info) => {
+    function compose(index: number): Response | Promise<Response> {
```

Comment:
Tiny nit: The name `compose` is a bit misleading, because this function doesn't compose anything. What we call `createHandler` here resembles much more the signature of a `compose` function.

Reply:
I've renamed `createHandler` to `composeHandler`.